creator-node/src/routes/tracks.js

```diff
@@ -24,169 +21,39 @@ const {
-        TranscodingQueue.segment(req.fileDir, req.fileName, { logContext: req.logContext }),
-        TranscodingQueue.transcode320(req.fileDir, req.fileName, { logContext: req.logContext })
-      ])
-      segmentFilePaths = transcode[0].filePaths
-      transcodedFilePath = transcode[1].filePath
-
-      req.logger.info(`Time taken in /track_content to re-encode track file: ${Date.now() - codeBlockTimeStart}ms for file ${req.fileName}`)
-    } catch (err) {
-      // Prune upload artifacts
-      removeTrackFolder(req, req.fileDir)
-
-      return errorResponseServerError(err)
-    }
-
-    // Save transcode and segment files (in parallel) to ipfs and retrieve multihashes
-    codeBlockTimeStart = Date.now()
-    const transcodeFileIPFSResp = await saveFileToIPFSFromFS(req, transcodedFilePath)
-
-    let segmentFileIPFSResps = []
-    for (let i = 0; i < segmentFilePaths.length; i += SaveFileToIPFSConcurrencyLimit) {
-      const segmentFilePathsSlice = segmentFilePaths.slice(i, i + SaveFileToIPFSConcurrencyLimit)
-
-      const sliceResps = await Promise.all(segmentFilePathsSlice.map(async (segmentFilePath) => {
-        const segmentAbsolutePath = path.join(req.fileDir, 'segments', segmentFilePath)
-        const { multihash, dstPath } = await saveFileToIPFSFromFS(req, segmentAbsolutePath)
-        return { multihash, srcPath: segmentFilePath, dstPath }
-      }))
-
-      segmentFileIPFSResps = segmentFileIPFSResps.concat(sliceResps)
-    }
-    req.logger.info(`Time taken in /track_content for saving transcode + segment files to IPFS: ${Date.now() - codeBlockTimeStart}ms for file ${req.fileName}`)
-
-    // Retrieve all segment durations as map(segment srcFilePath => segment duration)
-    codeBlockTimeStart = Date.now()
-    const segmentDurations = await getSegmentsDuration(req.fileName, req.file.destination)
-    req.logger.info(`Time taken in /track_content to get segment duration: ${Date.now() - codeBlockTimeStart}ms for file ${req.fileName}`)
-
-    // For all segments, build array of (segment multihash, segment duration)
-    let trackSegments = segmentFileIPFSResps.map((segmentFileIPFSResp) => {
-      return {
-        multihash: segmentFileIPFSResp.multihash,
-        duration: segmentDurations[segmentFileIPFSResp.srcPath]
-      }
-    })
-
-    // exclude 0-length segments that are sometimes outputted by ffmpeg segmentation
-    trackSegments = trackSegments.filter(trackSegment => trackSegment.duration)
-
-    // error if there are no track segments
-    if (!trackSegments || !trackSegments.length) {
-      // Prune upload artifacts
-      removeTrackFolder(req, req.fileDir)
-
-      return errorResponseServerError('Track upload failed - no track segments')
-    }
+  app.post('/track_content', authMiddleware, ensurePrimaryMiddleware, ensureStorageMiddleware, syncLockMiddleware, handleTrackContentUpload, handleResponse(async (req, res) => {
+    const fileUploadError = req.fileSizeError || req.fileFilterError
```

Comment:
Nit - we could probably have the logic for handling these errors here as opposed to in the worker. Seems redundant to serialize and send it over when we can just send the response early here

Reply:
done